src/RE/B/BSLightingShaderMaterialBase.cpp

```
@@ -43,68 +43,69 @@ namespace RE
 
 	BSLightingShaderMaterialBase* BSLightingShaderMaterialBase::CreateMaterial(Feature a_feature)
 	{
+		auto scrapHeap = MemoryManager::GetSingleton()->GetThreadScrapHeap();
 		switch (a_feature) {
 		case Feature::kDefault:
 			{
-				auto material = malloc<BSLightingShaderMaterial>();
-				std::memset((void*)material, 0, sizeof(BSLightingShaderMaterial));
+				auto material = static_cast<BSLightingShaderMaterial*>(scrapHeap->Allocate(sizeof(BSLightingShaderMaterial), 8));
 				if (material) {
+					std::memset((void*)material, 0, sizeof(BSLightingShaderMaterial));
```

Cppcheck (2.19.0)

[error] Using 'memset' on class that contains a virtual function.

(memsetClass)